### ISSUE TITLE
Option for (real) fast scrolling (aka immediate jump)

### ISSUE THREAD

**Issue by dehmer** [wellyshen/react-cool-virtual#243]
# Feature Request

## Describe the Feature

Is it possible to achieve more or less immediate scroll/jump to item with index?
It would be ideal to scroll fast from top to bottom (or anywhere in-between) and back.
Note: Scrolling from bottom to top already behaves like expected.

## Describe the Solution You'd Like

It would be nice, if scrollToItem() would resemble [Element.scrollIntoView()](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView). Specifically with { behavior: 'auto' }. Supporting `block` and `inline` options would be perfect, but (very) optional.

## Describe Alternatives You've Considered

N/A. Currently I'm relying on Element.scrollIntoView() for non-virtualised lists.

## Additional Information

The following CodeSandbox demonstrates an issue with scrolling from top to the bottom of the list: https://codesandbox.io/s/white-glade-xnu3u

1. Focus list element
2. Press `End` key
3. Wait
4. Wait some more, ..., wait for it, ..., yes, yes, nearly done...
5. Got it, finally!

**Comment by dehmer** [comment]
Sorry, I forgot to mention, that **variable height** items are imperative for our usage scenarios.
I can imagine that this poses an additional challenge. - Or makes it impossible to achieve.

**Comment by wellyshen** [comment]
@dehmer Remove the `max-height: 0` property from the `list` class then try it again.

**Comment by dehmer** [comment]
@wellyshen Sweet Jesus! You're the man!
This was a leftover workaround from before I found `height: calc(100vh - 20em)` to adjust panel heights with window.
Like I've said, I'm a CSS noob.

BTW: This fixes also the issue that scrollbar size was permanently adjusted while scrolling.

I think, we are pretty close of putting your excellent hook into production. 
This would relief us of a major headache.

Maybe you are interested why (might not belong here, but what the heck!):
1. Alternative libraries have a hard time dealing with variable heights and are generally not on par with your work.
2. Alternative libraries have lousy interfaces, often times assuming too much of the underlying list model.
3. We cannot simply truncate the lists, because there are use cases requiring bulk operations on ALL elements in the list (visible or not).

On that note, we are considering to make a donation to you Open Collective. 
I already talked to our 'CFO' and he's good with it.

Closing as fixed.
